app/src/main/java/com/github/sdp/tarjetakuna/database/UserCardsRTDB.kt

```diff
@@ -60,4 +69,21 @@ class UserCardsRTDB {
         }
         return future
     }
+
+    /**
+     * Retrieves all the cards asynchronously from the database
+     */
+    fun getAllCardsFromCollection(): CompletableFuture<DataSnapshot> {
+        val future = CompletableFuture<DataSnapshot>()
+        userCardCollection?.get()?.addOnSuccessListener {

```

Comment:
What's the type of `userCardCollection?.get()?` here? It seems odd you would have to manually create a new future and do the complete/completeExceptionally stuff yourself...